modules/apps/29-fee/keeper/msg_server.go

```diff
@@ -1,4 +1,37 @@
 package keeper
 
-// TODO
-//var _ types.MsgServer = Keeper{}
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/modules/apps/29-fee/types"
+)
+
+var _ types.MsgServer = Keeper{}
+
+// RegisterCounterpartyAddress is called by the relayer on each channelEnd and allows them to specify their counterparty address before relaying
+// This ensures they will be properly compensated for forward relaying since destination chain must send back relayer's source address (counterparty address) in acknowledgement
+// This function may be called more than once by relayer, in which case, latest counterparty address is always used.
+func (k Keeper) RegisterCounterpartyAddress(goCtx context.Context, msg *types.MsgRegisterCounterpartyAddress) (*types.MsgRegisterCounterpartyAddressResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := msg.ValidateBasic(); err != nil {
```

Comment:
I'm guessing that ValidateBasic does not need to be called here? From what I understand this gets called by `CheckTx`, correct?

I just left this in to ask the question.